pkg/services/login/login.go

```diff
@@ -0,0 +1,245 @@
+package login
```

Comment:
> Can you split the move of this service out into it's own PR? We should try not to do unrelated changes in PRs to make changes easier to follow and understand.

I wish! The thing is, the ldap indirectly depends on some of the parts of the login - https://github.com/grafana/grafana/blob/57dfea9b24c3d30002dc186a24f8854a22da3d69/pkg/services/ldap/ldap.go#L230, which is fine but in tests we have more clear dependency - https://github.com/grafana/grafana/blob/57dfea9b24c3d30002dc186a24f8854a22da3d69/pkg/services/ldap/test.go#L71

And `pkg/login` already depends on ldap, so If I provide the pull without this separation I will get a circle dependency :/. I probably can workaround that in tests and a bit of further refactoring, but it seems it's just easier to tackle it this task right now.

> Also, we need to remove this code from ext_user.go when moving.

Yep, haven't finished chastising this module yet :)

Reply:
Right, well, in that case, we should definitely do the move now. Is there anything that stops you from removing the code from ext_user.go now? What do you mean by chastising? 

It would be preferable if the code was removed and added back within the same commit as that means git keeps a reference of the origin of the code. moving in one commit and then removing later makes the history harder to find.

Reply:
> What do you mean by chastising?

I mean "refactoring" here i.e. gutting it, ransacking and eviscerating it :). I haven't finished doing that so there is some duplication, there was much more when the pull was introduced :).

> It would be preferable if the code was removed and added back within the same commit as that means git keeps a reference of the origin of the code. moving in one commit and then removing later makes the history harder to find.

Do we want to split it up? Or? Since after merging it it will be one commit

Reply:
Ah!

No, you're right. Just remove it and the squash will take care of the reference problem at merge.